csharp-demo/src/HelloWorldFunction/Function.cs

```diff
@@ -0,0 +1,44 @@
+using System.Runtime.InteropServices;
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+
+namespace HelloWorldFunction;
+
+public class Function
+{
+    public Function()
+    {
+        PosixSignalRegistration.Create(PosixSignal.SIGTERM, SignalHandler);
+    }
+    /// <summary>
+    /// A simple function that takes a string and does a ToUpper
+    /// </summary>
+    /// <param name="request"></param>
+    /// <param name="context"></param>
+    /// <returns></returns>
+    public string? FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context)
+    {   
+        return request.Body?.ToUpper();
```

Comment:
This response does not match API Gateway proxy response format. Could be updated as something like below. 

```csharp
    public APIGatewayProxyResponse FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context)
    {   
        return new APIGatewayProxyResponse{
            StatusCode = 200,
            Body = "Hello world!",
        };
         
    }
```